### PR TITLE
Pretty diagnostic output

### PR DESCRIPTION
**Pull request by JeffAshton**

### CODE REVIEW

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 1]
```diff
@@ -12,29 +14,9 @@ namespace D2L.CodeStyle.Analyzers {
 	[TestFixtureSource( typeof( SpecTestsProvider ), nameof( SpecTestsProvider.GetAll ) )]
 	internal sealed class Spec {
 
-		/// <summary>
-		/// Compares diagnostics based on their Id and location
-		/// </summary>
-		private sealed class DiagnosticComparer : IEqualityComparer<Diagnostic> {
```
Comment (JeffAshton) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:18]: Rolled into `PrettyDiagnostic` generated `record` implementation:

https://github.com/Brightspace/D2L.CodeStyle/pull/848/files#diff-a24e955b566074869c45ab4ad8d1d7a9cd4556bf5d3b0b811f3129c42f6a7ea8R318

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 2]
```diff
@@ -49,27 +31,34 @@ private sealed class DiagnosticComparer : IEqualityComparer<Diagnostic> {
...
 		}
 
 		[Test]
 		public void NoUnexpectedDiagnostics() {
-			var unexpectedDiagnostics = m_actualDiagnostics
+
+			IEnumerable<PrettyDiagnostic> unexpectedDiagnostics = m_actualDiagnostics
 				.Where( d => !m_matchedDiagnostics.Contains( d ) );
 
+			Assert.Multiple( () => {
```
Comment (JeffAshton) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:56]: Using `Assert.Multiple` to gather all 

image

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 3]
```diff
@@ -124,8 +118,8 @@ ImmutableArray<AdditionalText> additionalFiles
 					analyzers: ImmutableArray.Create( analyzer ),
 					options: new AnalyzerOptions( additionalFiles )
 				)
-				.GetAnalyzerDiagnosticsAsync().Result
-				.ToImmutableArray();
```
Comment (JeffAshton) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:128]: `Result` is already an `ImmutableArray`

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 4]
```diff
@@ -320,5 +314,45 @@ ImmutableArray<MetadataReference> metadataReferences
...
+					sb.Append( " = " );
+					sb.Append( value );
+					sb.AppendLine( seperator );
+				}
+
+				sb.AppendLine( "{" );
+				WriteProperty( nameof( Id ), Id );
+				WriteProperty( nameof( LinePosition ), LinePosition );
+				WriteProperty( nameof( Message ), Message, seperator: string.Empty );
+				sb.AppendLine( "}" );
```
Comment (omsmith) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:344]: I could imagine adding the actual sourceline with an arrow pointing to the position in the future, so, cool setup.

**Review by omsmith — DISMISSED** [review]

File: tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs [hunk 5]
```diff
@@ -49,27 +31,34 @@ private sealed class DiagnosticComparer : IEqualityComparer<Diagnostic> {
...
-				).ToImmutableHashSet( DiagnosticComparer.Instance );
+				.Intersect( m_expectedDiagnostics )
+				.ToImmutableHashSet();
 		}
 
 		[Test]
 		public void NoUnexpectedDiagnostics() {
-			var unexpectedDiagnostics = m_actualDiagnostics
+
+			IEnumerable<PrettyDiagnostic> unexpectedDiagnostics = m_actualDiagnostics
```
Comment (j3parker) [inline tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs:53]: unnecessary changes

**Review by j3parker — DISMISSED** [review]

**Comment by omsmith** [comment]
That merge looks really weird

**Comment by JeffAshton** [comment]
> That merge looks really weird

Yup.  Shouldn't have squashed that last one

**Review by omsmith — APPROVED** [review]
